### PR TITLE
DO NOT MERGE, CI ONLY Downstream 0.5.0 03

### PR DESCRIPTION
**Pull request by avinal**
- Allow users/integrators to pass arbitrary keys/values through Result and RecordSummary annotations.
- Fix version tag for results release
- Update kustomization files to latest standard
- Bump gorm.io/gorm from 1.24.3 to 1.25.0
- Bump go version to 1.19
- Make e2e tests more configurable
- Bump google.golang.org/grpc from 1.53.0 to 1.55.0
- Bump k8s.io/api from 0.25.8 to 0.25.9
- [CARRY]: Tekton Results downstream configuration

### CODE REVIEW

**Comment by avinal** [comment]
cc @prietyc123

File: .tekton/tekton-results-api-push.yaml [hunk 1]
```diff
@@ -0,0 +1,33 @@
+---
+# Triggers Tekton Results api image build on pull request
```
Comment (prietyc123) [inline .tekton/tekton-results-api-push.yaml:2]:
```suggestion
# Triggers Tekton Results api image build on push request
```

File: downstream.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+2. Fetch the branches from the downstream repo: `git fetch downstream` and find the most recent downstream branch (e.g `downstream/downstream-0.5.0-01`)
+3. Create a new branch from it: `git checkout -b <downstream-X.Y.Z-N(N+1)>  downstream/<downstream-X.Y.Z-NN>` (e.g `git checkout -b downstream-0.5.0-02  downstream/downstream-0.5.0-01`)
+4. If there are commits on top of the "[CARRY]: Tekton Results downstream configuration" commit, squash them preserving the "[CARRY]: Tekton Results downstream configuration" commit message. There should be only one downstream commit!
+5. Rebase the main branch: `git rebase main`
+6. Push the branch to your fork: `git push origin <downstream-X.Y.Z-NN>` (e.g `git push origin downstream-0.5.0-02`)
+7. Open a PR choosing base repository "openshift-pipeline/tektoncd-results" and base branch the latest downstream-X.Y.Z-NN branch (here downstream-0.5.0-01).
+    1. Add "DO NOT MERGE, CI ONLY" prefix to the PR title
+    2. In the description add and properly update the following fields:
```
Comment (prietyc123) [inline downstream.md:39]:
```suggestion
    2. In the description add and properly update the following fields:
```
Could you please elaborate this line more.

**Review by prietyc123 — CHANGES_REQUESTED** [review]
Minor changes requested.

**Comment by avinal** [comment]
Will create another PR in favor of 0.6.0, closing this.

File: downstream.md (hunk 2 above)
Comment (enarha) [reply downstream.md:39]: @prietyc123 when creating the PR for testing the mix of upstream + downstream changes, we edit the PR description and add the fields (lines) listed below, this is what "add" stands for. We also "update" them to reflect on the upstream changes included in the PR, specifically which ones are new features (if any), changes which require updates in the deployment configuration and breaking changes.

**Pull request closed without merging**